cosmwasm/lib/wasmi-runtime/Cargo.toml

```diff
@@ -5,7 +5,43 @@ authors = ["Reuven Podmazo <reuven@enigma.co>"]
 edition = "2018"
 description = "An enclave running wasmi, to be used by cosmwasm-sgx-vm"
 
+[lib]
+name = "wasmi_runtime_enclave"
+crate-type = ["staticlib"]
+
 [dependencies]
 enclave-ffi-types = { path = "../enclave-ffi-types" }
+wasmi = { git = "https://github.com/paritytech/wasmi", branch = "master", default-features = false, features = ["core"] }
+
+[features]
+default = []
 
-[build-dependencies]
+[target.'cfg(not(target_env = "sgx"))'.dependencies]
+sgx_types = { git = "https://github.com/apache/teaclave-sgx-sdk.git" }
+sgx_tstd = { git = "https://github.com/apache/teaclave-sgx-sdk.git", features = ["backtrace"] }
+sgx_trts = { git = "https://github.com/apache/teaclave-sgx-sdk.git" }
+[patch.'https://github.com/apache/teaclave-sgx-sdk.git']
+sgx_alloc = { path = "../incubator-teaclave-sgx-sdk/sgx_alloc" }
+sgx_build_helper = { path = "../incubator-teaclave-sgx-sdk/sgx_build_helper" }
+sgx_cov = { path = "../incubator-teaclave-sgx-sdk/sgx_cov" }
+sgx_crypto_helper = { path = "../incubator-teaclave-sgx-sdk/sgx_crypto_helper" }
+sgx_libc = { path = "../incubator-teaclave-sgx-sdk/sgx_libc" }
+sgx_rand = { path = "../incubator-teaclave-sgx-sdk/sgx_rand" }
+sgx_rand_derive = { path = "../incubator-teaclave-sgx-sdk/sgx_rand_derive" }
+sgx_serialize = { path = "../incubator-teaclave-sgx-sdk/sgx_serialize" }
+sgx_serialize_derive = { path = "../incubator-teaclave-sgx-sdk/sgx_serialize_derive" }
+sgx_serialize_derive_internals = { path = "../incubator-teaclave-sgx-sdk/sgx_serialize_derive_internals" }
+sgx_tcrypto = { path = "../incubator-teaclave-sgx-sdk/sgx_tcrypto" }
+sgx_tcrypto_helper = { path = "../incubator-teaclave-sgx-sdk/sgx_tcrypto_helper" }
+sgx_tdh = { path = "../incubator-teaclave-sgx-sdk/sgx_tdh" }
+sgx_tkey_exchange = { path = "../incubator-teaclave-sgx-sdk/sgx_tkey_exchange" }
+sgx_tprotected_fs = { path = "../incubator-teaclave-sgx-sdk/sgx_tprotected_fs" }
+sgx_trts = { path = "../incubator-teaclave-sgx-sdk/sgx_trts" }
+sgx_tse = { path = "../incubator-teaclave-sgx-sdk/sgx_tse" }
+sgx_tseal = { path = "../incubator-teaclave-sgx-sdk/sgx_tseal" }
+sgx_tstd = { path = "../incubator-teaclave-sgx-sdk/sgx_tstd" }
+sgx_tunittest = { path = "../incubator-teaclave-sgx-sdk/sgx_tunittest" }
+sgx_types = { path = "../incubator-teaclave-sgx-sdk/sgx_types" }
+sgx_ucrypto = { path = "../incubator-teaclave-sgx-sdk/sgx_ucrypto" }
+sgx_unwind = { path = "../incubator-teaclave-sgx-sdk/sgx_unwind" }
+sgx_urts = { path = "../incubator-teaclave-sgx-sdk/sgx_urts" }
```

Comment:
1. The first section there says you're only pulling these dependencies when NOT building for SGX.
2. You then override the github links with local paths.
3. You should be pulling only the dependencies you need, and you're not using most of these
4. All of these are available on crates.io. please just specify normal crates.io dependencies.

Reply:
We only copied this from the sample project, trying to make it work rn

Reply:
resolved in call